src/core/analysis-context.ts

```diff
@@ -496,4 +487,28 @@ export class AnalysisContext {
     return this.resolver.canResolve(url) ? this.resolver.resolve(url) :
                                            (url as any as ResolvedUrl);
   }
+
+  /**
+   * A warning for a weird situation that should not obtain.
+   *
+   * Before calling getDocument(), which is synchronous, a caller must first
+   * have finished loading and scanning, as those phases are asynchronous.
+   *
+   * So we need to construct a warning, but we don't have a parsed document.
```

Comment:
It'd be nice if either 1) warnings didn't require documents, or 2) we produced a warning on the initiator document

Reply:
re 2), for most warnings we do, including parse errors, but for this "impossible" warning we don't even have that much info

I'd like to hold off on 1)